update to v2.23.2

hideyosh1 opened the pull request:
i also updated the runtimes, cups, shared-modules -- idk why cups is necessary, consider removing it? added wayland support also. lmk if there's anything weird

hideyosh1 committed:
update cups and gtk and freedesktop runtime, v2.23.2, remove libsecret with new electron

allow wayland with ozone -- allow ime support

remove libsecret dependency & associated gitmodules

hideyosh1 commented:
@mfschumann @rhjhunt ?

mfschumann commented:
I tried your latest test build on Fedora 42 with Gnome 48 on Wayland, but starting the flatpak gets stuck after the version check:
```
[2:0424/192113.423620:ERROR:bus.cc(407)] Failed to connect to the bus: Failed to connect to socket /run/dbus/system_bus_socket: No such file or directory
LaunchProcess: failed to execvp:
xdg-settings
[2:0424/192114.284641:ERROR:bus.cc(407)] Failed to connect to the bus: Failed to connect to socket /run/dbus/system_bus_socket: No such file or directory
[2:0424/192114.284697:ERROR:bus.cc(407)] Failed to connect to the bus: Failed to connect to socket /run/dbus/system_bus_socket: No such file or directory
Checking for update
Update for version 2.23.2 is not available (latest version: 2.23.2, downgrade is disallowed).
```
When running with `--log-session-bus` I cannot see any missing permissions.

mfschumann commented:
Running with `--socket=system-bus` makes the `Failed to connect to the bus` errors go away, but does not make a difference otherwise, i.e., startup gets stuck and no window appears.

hideyosh1 commented:
hmmm... the "system bus permission denied" is a thing i see in a ton of electron apps and i've always thought it was just an "electron thing?" i added it just in case. but anyways i think the problem was that i removed the cups dependency because it looked like it worked on my machine without problems. can you check if it works now?

ps - the cursor themes i use are completely broken and idk really what to do about that...

mfschumann commented:
I have the same startup issue still.

mfschumann commented:
Btw, I had a look at the changes you made; you removed quite some stuff:

- access to `xdg-config/Simplenote`
- `gtk-settings.ini` which sets the printing providers
- `libsecret`

Is all of that not needed anymore?

mfschumann commented on com.simplenote.Simplenote.desktop:
```diff
@@ -1,6 +1,6 @@
 [Desktop Entry]
 Name=Simplenote
-Exec=simplenote %U
+Exec=simplenote --enable-features=UseOzonePlatform --ozone-platform-hint=auto  --enable-wayland-ime --wayland-text-input-version=3 %U
```
It's probably better to put these arguments in a wrapper script that is executed when running the flatpak. Like that, the arguments are also used when simplenote is started via `flatpak run` on the command line, not only when started via the `.desktop` file.

hideyosh1 commented:
according to #9 libsecret shouldn't be needed anymore --
i don't think i deleted gtk-settings.ini or if i did i probably fixed it in a later commit and brought it back (sorry for the mess)
access to xdg-config/Simplenote shouldn't be necessary because i thought it was automatically binded to ~/.var/app/com.simplenote.Simplenote/...? and the linter was yelling at me for that - https://docs.flathub.org/docs/for-app-authors/linter#finish-args-unnecessary-xdg-config-subdir-mode-access ?

hideyosh1 commented:
i'll try to see what's up with the startup issue tomorrow, but that's rly weird i'm on gnome 47.4 nixos and it's been alright...i'm going to try to run in a fedora vm tomorrow to try also

hideyosh1 committed:
fix cursor customization with XCURSOR_PATH

mfschumann commented:
I tried the latest test build, but that still has the startup issue for me.

hideyosh1 commented:
yes, I also tested it on a fedora vm and it seemed to be as a result of some upstream electron issue because I know other apps like logseq and vivaldi have trouble launching  [like in this report](https://discussion.fedoraproject.org/t/applications-and-flatpaks-fail-to-launch-flatpak-launchprocess-failed-to-execvp-xdg-settings/122997)

I'll try this in another distro to see if this is reproducible in ubuntu or something like that

MoralCode commented:
i feel like ive seen a similar system bus access pattern when trying to run an X11 flatpak in a wayland only environment. sounds like that may be the issue? (if im not totally off topic/on a different page that is)

mfschumann commented:
bot, build

ghost commented:
Outdated. Closed.
